File: Hilbert - Unity/Assets/Scripts/LearnManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;
using TMPro;
using System.Linq;

public class LearnManager : MonoBehaviour
{
    [System.Serializable]
    public class StoneData
    {
        public string xName;
        [Space(10)]
        [TextArea(6, 10)]
        public string xLore;
        [Space(10)]
        [TextArea(10,40)]
        public string xGuide;
        [Space(10)]
        public Sprite xStoneImage;
    }


    public StoneData[] stonedb;
    private static List<StoneData> stonelib;
    [Space(50)]

    public TextMeshProUGUI LoreSlot;
    public TextMeshProUGUI GuideSlot;
    public TextMeshProUGUI indexSlot;
    public TextMeshProUGUI titleSlot;
    public Image StoneImageSlot;
    private int currentIndex = 1;
    [Space(10)]

    public GameObject backButton;
    public GameObject nextButton;
    public Button StoneDataButton;
    public void Start()
    {
        stonelib = stonedb.ToList<StoneData>();

        StoneDataButton.interactable = false;

    }
    public void Update()
    {
        DisplayStone();
        CheckEnd();
    }

    private void CheckEnd()
    {
        if (currentIndex == 1)
        {
            backButton.SetActive(false);

        }
        else
        {
            backButton.SetActive(true);
        }

        if(currentIndex == stonedb.Length)
        {
            nextButton.SetActive(false);
        }
        else
        {
            nextButton.SetActive(true);
        }

    }
    private void DisplayStone()
    {
        indexSlot.text = currentIndex.ToString();
        titleSlot.text = stonelib[currentIndex - 1].xName;
        LoreSlot.text = stonelib[currentIndex - 1].xLore;
        GuideSlot.text = stonelib[currentIndex - 1].xGuide;
        StoneImageSlot.sprite = stonelib[currentIndex - 1].xStoneImage;
    }

    public void AddIndex()
    {
        currentIndex += 1;
    }

    public void SubIndex()
    {
        currentIndex -= 1;
    }
}

File: Hilbert - Unity/Main/Scripts/LearnManager.py
import UnityEngine as ue

LoreSlot = ""
InstructionSlot = ""
IndexSlot = ""
TitleSlot = ""
StoneImageSlot = ue.Image(None)
currentIndex = 1

backButton = ue.Button() #manual selection
nextButton = ue.Button() #manual selection
StoneDataButton = ue.Button() #manual selection

class StoneLearnData:
    def __init__(self, xName, xLore, xInstruction, xStoneImage):
        self.xName = xName
        self.xLore = xLore
        self.xInstruction = xInstruction
        self.xStoneImage = xStoneImage

stonedb = gameObject.StoneLearnData("defaultName", "defaultLoreText", "defaultGuideText", ue.Image()) #Manully placed in editor

class LearnManager():
    def __init__(self, LoreSlot, InstructionSlot, IndexSlot, TitleSlot, StoneImageSlot, currentIndex, backButton, nextButton, StoneDataButton):
        self.LoreSlot = LoreSlot
        self.InstructionSlot = InstructionSlot
        self.IndexSlot = IndexSlot
        self.TitleSlot = TitleSlot
        self.StoneImageSlot = StoneImageSlot
        self.currentIndex = currentIndex
        self.backButton = backButton
        self.nextButton = nextButton
        self.StoneDataButton = StoneDataButton

    def CheckEnd():
        if(currentIndex == 1): backButton.SetActive = False
        else: backButton.SetActive = True
        if(currentIndex == len(stonedb)): nextButton.SetActive = False
        else: nextButton.SetActive = True

    def DisplayStone():
        IndexSlot = str(currentIndex)
        TitleSlot = stonedb[currentIndex-1].xName
        LoreSlot = stonedb[currentIndex-1].xLore
        InstructionSlot = stonedb[currentIndex-1].xInstruction
        StoneImageSlot = stonedb[currentIndex-1].xStoneImage

    def Start(self):
        StoneDataButton.interactable = False

    def Update(self):
        DisplayStone()
        CheckEnd()
